### PR TITLE
Performance: String papercuts

### PR DESCRIPTION
**Pull request by NickCraver**
.NET 5-6 has some enhancements on string handling we can take advantage of...and just some cleanup along the way. I'd like to get analyzers running on these long-term, so cleaning up on the way to proposing such a thing to help us keep it optimized. These aren't going to be large needle moves on perf (some are zero, just tidy), only more things that add up.

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerActivityPublisher.cs [hunk 1]
```diff
@@ -165,7 +165,7 @@ private async Task FlushFunctionExecutionActivities()
             }
             catch (Exception exc) when (!exc.IsFatal())
             {
-                _logger.LogError(exc, $"{nameof(FlushFunctionExecutionActivities)}");
+                _logger.LogError(exc, nameof(FlushFunctionExecutionActivities));
```
Comment (fabiocav) [inline src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerActivityPublisher.cs:168]: 😕

**Review by fabiocav — APPROVED** [review]

**Review by liliankasem — APPROVED** [review]
🚀

File: src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerActivityPublisher.cs (hunk 1 above)
Comment (NickCraver) [reply src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerActivityPublisher.cs:168]: Is it okay to change/improve these? Thus far I've been very wary of breaking changes, but happy to help improve the errors (I haven't yet learned which ones customers see vs. not)

Comment (liliankasem) [reply src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerActivityPublisher.cs:168]: I read that emoji as "I can't believe we had that code in there" 😆

**Pull request merged**